tests/test_mutations.py

```diff
@@ -339,6 +339,119 @@ def test_new_model(self):
         self.validate_model(model)
         self.validate_stationary(model)
 
+    def test_SMM(self):
+        model = msprime.SMM(lo=2, hi=50)
+        # test for symmetry of SMM trans mat
+        for i in range(model.transition_matrix.shape[0]):
+            for j in range(i):
+                if i != j:
+                    assert (
+                        model.transition_matrix[i, j] == model.transition_matrix[j, i]
+                    )
+                if np.abs(i - j) > 1:
+                    assert model.transition_matrix[i, j] == 0.0
+
+        self.validate_model(model)
+        self.validate_stationary(model)
+
+    def test_TPM(self):
+        lo = 2
+        hi = 50
+        m = 0.9
+        p = 0.9
+        model = msprime.TPM(p=p, m=m, lo=lo, hi=hi)
+        self.validate_model(model)
+        self.validate_stationary(model)
+
+        # test for ratio of TPM trans mat vals
+        for i in range(lo, hi + 1):
+            if lo < i < hi:
+                ii = i - lo
+                exp = (p + (m / (1 - (1 - m) ** (hi - i))) * (1 - p)) / (
+                    p + (m / (1 - (1 - m) ** (i - lo)) * (1 - p))
+                )
+                obs = (
+                    model.transition_matrix[ii, ii + 1]
+                    / model.transition_matrix[ii, ii - 1]
+                )
+                assert np.isclose(exp, obs)
+
+    def test_EL2(self):
+        model = msprime.EL2()
+        self.validate_model(model)
+        self.validate_stationary(model)
+
+        lo = 1
+        hi = 10
+        m = 0.43
+        u = 0.68
+        v = 0.037
+        p = 0
+        model = msprime.EL2(lo=lo, hi=hi)
+        for i in range(lo, hi + 1):
+            if lo < i < hi:
+                ii = i - lo
+                g_ratio = (p + (m / (1 - (1 - m) ** (hi - i))) * (1 - p)) / (
+                    p + (m / (1 - (1 - m) ** (i - lo)) * (1 - p))
+                )
+                a_ratio = u - (v * (i - lo))
+                a_ratio /= 1 - a_ratio
+                exp = g_ratio * a_ratio
+                obs = (
+                    model.transition_matrix[ii, ii + 1]
+                    / model.transition_matrix[ii, ii - 1]
+                )
+                if np.abs(exp - obs) > 1e-6:
+                    raise ValueError(f"failed test on row: {i} exp:{exp} obs:{obs}")
```

Comment:
Whoops - left over from debugging?
```suggestion
                assert np.abs(exp - obs) < 1e-6
```

Reply:
yup